serving-core/src/main/scala/com/stratio/sparkta/serving/core/actor/FragmentActor.scala

```diff
@@ -89,22 +89,36 @@ class FragmentActor(curatorFramework: CuratorFramework)
 
   def create(fragment: FragmentElementModel): Unit =
     sender ! ResponseFragment(Try({
-      if (existsByTypeAndName(fragment.fragmentType, fragment.name.toLowerCase)) {
-        throw new ServingCoreException(ErrorModel.toString(
-          new ErrorModel(ErrorModel.CodeExistsFragmentWithName,
-            s"Fragment of type ${fragment.fragmentType} with name ${fragment.name} exists.")
-        ))
+      val maybeExistingFragment: Option[FragmentElementModel] = existsByTypeAndName(fragment.fragmentType, fragment.name
+        .toLowerCase) match {
+        case true => getExistingFragment(fragment)
+        case _    => None
+      }
+      maybeExistingFragment.isDefined match {
```

Comment:
Other option is to use a method getOrElse. IMHO this method match very well in this case:
```scala
maybeExistingFragment.getOrElse(createNewFragment(fragment))

// ...

def createNewFragment(fragment: FragmentElementModel): Fragment = {
 val fragmentS = fragment.copy(
   id = Option(UUID.randomUUID.toString),
   name = fragment.name.toLowerCase
  )
  curatorFramework.create().creatingParentsIfNeeded().forPath(
    s"${FragmentActor.fragmentPath(fragmentS.fragmentType)}/${fragmentS.id.get}", write(fragmentS).getBytes()
  )
 fragmentS
}
```